### PR TITLE
Convert to using pyproject.toml

### PR DESCRIPTION
**Pull request by plars**
This updates testflinger-agent to use the newer pyproject.toml setup methods. A minimal setup.py is preserved for backwards compatibility purposes

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env python
```
Comment (yphus) [inline setup.py]: #!/usr/bin/env python3

**Review by yphus — CHANGES_REQUESTED** [review]
LGTM, in particular it makes this old line `setup.py testflinger-agent testflinger_agent` much more understandable.
One tiny fix though to the python interpreter set for `setup.py`, see below

File: setup.py (hunk 1 above)
Comment (plars) [reply setup.py]: Thanks, fixed!

**Review by kissiel — APPROVED** [review]
LGTM, +1
I think this is a very motivating and trailblazing PR.

**Pull request merged**